cmd/otelcol/main.go

```diff
@@ -291,11 +290,6 @@ func setMemoryLimit(memTotalSizeMiB int) {
 	// If not, calculate it from memTotalSizeMiB
 	if memoryLimit == "" {
 		memLimit = memTotalSizeMiB * defaultMemoryLimitPercentage / 100
-		// The memory limit should be set to defaultMemoryLimitPercentage of total memory
-		// while reserving a maximum of defaultMemoryLimitMaxMiB of memory.
-		if (memTotalSizeMiB - memLimit) > defaultMemoryLimitMaxMiB {
-			memLimit = defaultMemoryLimitMaxMiB
```

Comment:
Keeping logic you can just change this to:

```
memLimit = (memTotalSizeMiB - defaultMemoryLimitMaxMiB)
```

Reply:
Yes, we discussed that option with @pjanotti and agreed on removing the logic.